pyfr/integrators/dual/phys/controllers.py

```diff
@@ -47,8 +51,21 @@ def advance_to(self, t):
             raise ValueError('Advance time is in the past')
 
         while self.tcurr < t:
+
+            if self.tcurr + 2*self._dt < t < self.tcurr + 3*self._dt:
```

Comment:
I think this is a way to ensure somewhat smoother transition down to the required level, but not sure if this is the best way. It looks like this will decrease the dt first and then increase a bit in some cases. Can you tabulate how the dt changes for a few different scenarios?

Reply:
And if your physical time stepper can handle this level of dt and the subsequent dtau change without introducing any error it means that the rest of stuff you're doing with the dtau works very well!

Reply:
> Can you tabulate how the dt changes for a few different scenarios?

That question actually made me get a bug. I was over-shooting the `dt` by about 1.5 times to hit the desired time. This is how dt varied with time for fixed case. We want this to be lesser than the set `dt` value. 

![variable-dt_fixed](https://user-images.githubusercontent.com/63449209/217327243-bccfb25e-a3cd-492b-ae5d-10c60ae5355a.png)

This is how the dt varied when I added random dt at every step.

![variable-dt_random](https://user-images.githubusercontent.com/63449209/217326258-6c4046e3-771e-43b5-afa5-005d00e5e4e9.png)

I just fixed the bug where `dt` is overshooting the config-set value.